### PR TITLE
Add children getter with out

### PR DESCRIPTION
**Pull request by SignorPipo**

### CODE REVIEW

File: src/wonderland.ts [hunk 1]
```diff
@@ -3405,10 +3405,27 @@ export class Object3D {
...
-            this.objectId
-        );
-        if (childrenCount === 0) return [];
+        return this.getChildren();
+    }
+
+    /**
+     * The number of children of this object.
+     */
+    getChildrenCount(): number {
```
Comment (DavidPeicho) [inline src/wonderland.ts:3414]: This could be a getter since it doesn't require an argument

Comment (DavidPeicho) [reply src/wonderland.ts:3414]: @SignorPipo This isn't resolved

**Comment by DavidPeicho** [comment]
Could you please add the same functions in `scene.ts` as well?

File: src/wonderland.ts (hunk 1 above)
Comment (SignorPipo) [reply src/wonderland.ts]: I forgot to push LOL

**Comment by SignorPipo** [comment]
> Could you please add the same functions in `scene.ts` as well?

What do you mean? Why and where specifically?

**Comment by DavidPeicho** [comment]
> > Could you please add the same functions in `scene.ts` as well?
> 
> What do you mean? Why and where specifically?

`scene.ts` now has similar functions that run on the root `get children` etc... I just merged the latest api into the main branch

File: src/wonderland.ts [hunk 2]
```diff
@@ -3933,12 +3949,17 @@ export class Object3D {
      *
      * When the object exists in the scene at editor time, prefer passing it as
      * a component property.
+     *
+     * If you use the out parameter, the returned array might have more elements than the actual children count,
```
Comment (DavidPeicho) [inline src/wonderland.ts:3953]: I think a slightly better phrasing would be:

```js
/**
 * @note When providing an output array, only `this.childrenCount` elements will be written.
 * The rest of the array will not be modified by this method.
 */
```

File: src/scene.ts [hunk 3]
```diff
@@ -254,6 +254,43 @@ export class Scene {
...
+     * Children of the root object.
+     * If you use the out paramter, the returned array might have more elements than the actual children count,
+     * if the out array was bigger than the actual childre count, so you should rely
+     * on the this.childrenCount value when iterating on it.
+     *
+     * @param out Destination array, expected to have at least this.childrenCount elements.
+     * @returns The `out` parameter.
+     */
+    getChildren(out: Object3D[] = new Array(this.childrenCount)): Object3D[] {
+        const childrenCount = this.childrenCount;
```
Comment (DavidPeicho) [inline src/scene.ts:275]: A faster to write implementation is to wrap the object and forward the call to `getChildren`:|

```ts
const root = this._engine.wrapObject(0);
return root.getChildren(...);
```

File: src/scene.ts [hunk 4]
```diff
@@ -254,6 +254,43 @@ export class Scene {
         return root.children;
```
Comment (DavidPeicho) [inline src/scene.ts:254]: While that's equivalent, this method should probably also directly use `this.getChildren()`

File: src/scene.ts [hunk 5]
```diff
@@ -254,6 +254,43 @@ export class Scene {
...
+        return this._engine.wasm._wl_object_get_children_count(rootObjectId);
+    }
+
+    /**
+     * Children of the root object.
+     * If you use the out paramter, the returned array might have more elements than the actual children count,
+     * if the out array was bigger than the actual childre count, so you should rely
+     * on the this.childrenCount value when iterating on it.
+     *
+     * @param out Destination array, expected to have at least this.childrenCount elements.
```
Comment (DavidPeicho) [inline src/scene.ts:271]: Use backtick around `this.childrenCount` please

File: src/scene.ts [hunk 6]
```diff
@@ -251,7 +251,29 @@ export class Scene {
...
+
+    /**
+     * The number of children of the root object.
+     */
+    get childrenCount(): number {
+        const rootObjectId = 0;
+        return this._engine.wasm._wl_object_get_children_count(rootObjectId);
+    }
+
+    /**
```
Comment (DavidPeicho) [inline src/scene.ts:265]: Maybe point this doc to the object's one:

```ts
    /**
     * Root object's children.
     *
     * See {@link Object3D.getChildren} for more information.
     *
     * @param out Destination array, expected to have at least `this.childrenCount` elements.
     * @returns The `out` parameter.
     */
```

**Comment by DavidPeicho** [comment]
Tiny comment and LGTM!

**Pull request merged**